sonar-plugin-api/src/main/java/org/sonar/api/component/Component.java

```diff
@@ -21,6 +21,10 @@
 
 import javax.annotation.CheckForNull;
 
+/**
+ * @deprecated since 5.2
+ */
+@Deprecated
 public interface Component {
```

Comment:
![New](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/new.png)![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png)Do not forget to remove this deprecated code someday.[![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)